pkg/objects/datatypes.go

```diff
@@ -25,13 +25,46 @@ import (
 	"errors"
 	"fmt"
 	"io/ioutil"
+	"os"
 	"regexp"
 	"strconv"
 	"time"
 
 	"github.com/ulikunitz/xz"
 )
 
+//AuthPassword contains the password for some OpenStack Authentication credentials.
+type AuthPassword string
+
+//UnmarshalYAML implements the yaml.Unmarshaler interface.
+func (p *AuthPassword) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	//plain text password
+	var plainTextInput string
+	err := unmarshal(&plainTextInput)
+	if err == nil {
+		*p = AuthPassword(plainTextInput)
+		return nil
+	}
+
+	//retrieve password from the given environment variable key
+	var envVariableInput struct {
+		Key string `yaml:"fromEnv"`
+	}
+	err = unmarshal(&envVariableInput)
+	if err != nil {
+		return err
+	}
+
+	passFromEnv := os.Getenv(envVariableInput.Key)
+	if passFromEnv == "" {
+		return fmt.Errorf(`environment variable %q does not have a valid OpenStack Authentication Password`, envVariableInput.Key)
```

Comment:
I'd just say `environment variable %q is not set`. The current wording sounds like we're validating against a regex or something like that.

Reply:
Rest looks good.